IPython/html/static/base/js/dialog.js

```diff
@@ -20,6 +20,9 @@ define([
         var dialog_content = $("<div/>")
             .addClass("modal-content")
             .appendTo(dialog);
+        if(typeof(options.body) === 'string'){
+            options.body = $("<p/>").text(options.body)
```

Comment:
Yup, it only sanitizes string dialogs. HTML dialogs aren't touched, and won't be.